src/BookingStateWrapper/BookingStateWrapper.js

```diff
@@ -1,34 +1,54 @@
 // @flow
 
 import * as React from 'react';
-import { graphql } from 'react-relay';
+import { graphql, QueryRenderer } from 'react-relay';
```

Comment:
why plain `QueryRenderer` from react-relay? our `QueryRenderer` contains additional logic for handling logout after 403

Reply:
It's definitely possible, but that component requires a router, so I'll have to put the booking state wrapper into `Routes`, then

Reply:
* under router

Reply:
So, I did just that, on this separate branch: https://github.com/kiwicom/smart-faq/tree/refactor-booking-section-logic-3-with-custom-query-renderer

Reply:
Do you need some help?

Reply:
I created this PR to make sure it's working: https://github.com/kiwicom/smart-faq/pull/759

Reply:
> Do you need some help?

Yes please!

Let's merge #759 first. I'll also push something first, which will resolve all the other issues that remain.